### ISSUE TITLE
Deploy Cluster Fails on retriviing list of schedulables nodes

### ISSUE THREAD

**Issue by masterkorp** [openshift/openshift-ansible#7416]
#### Description

On a simple single master install the `deploy_cluster.yml` playbook fails.

```
[OSEv3:children]
masters
nodes
etcd

[OSEv3:vars]
# if your target hosts are Fedora uncomment this
#ansible_python_interpreter=/usr/bin/python3
openshift_deployment_type=origin
openshift_release=3.7
osm_cluster_network_cidr=10.128.0.0/14
openshift_portal_net=172.30.0.0/16
osm_host_subnet_length=9
# localhost likely doesn't meet the minimum requirements
openshift_disable_check=disk_availability,memory_availability

[masters]
devops-1 ansible_host=10.0.14.90

[etcd]
devops-1 ansible_host=10.0.14.90

[nodes]
devops-1 ansible_host=10.0.14.90 openshift_schedulable=true

devops-3 ansible_host=10.0.14.92 openshift_schedulable=true openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
devops-4 ansible_host=10.0.14.93 openshift_schedulable=true openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
```

##### Version
```
ansible: 2.4.3.0
git describe: openshift-ansible-3.10.0-0.1.0-37-gf7700b5
```

##### Steps To Reproduce

```
ansible-playbook -u user-b -i inventory/hosts.devops-poc playbooks/pre_requesites.yml -vv
ansible-playbook -u user-b -i inventory/hosts.devops-poc playbooks/deploy_cluster.yml -vv
````

##### Expected Results

Should deploy

##### Observed Results

An oc command its being called with an option`--field-selector` that itss not being recgnized by `oc 

```
TASK [openshift_master : Retrieve list of schedulable nodes matching selector] *****************************************************************************************************************************************************************task path: /projects/openshift-ansible/roles/openshift_master/tasks/ensure_nodes_matching_selector.yml:2
Wednesday 07 March 2018  09:35:48 +0000 (0:00:00.329)       0:15:37.571 *******
fatal: [devops-1]: FAILED! => {"changed": false, "msg": {"cmd": "/bin/oc get node --selector=region=infra --field-selector=spec.unschedulable!=true -o json -n default", "results": [{}], "returncode": 1, "stderr": "Error: unknown flag: --field-selector\n\n\nUsage:\n  oc get [(-o|--output=)json|yaml|wide|custom-columns=...|custom-columns-file=...|go-template=...|go-template-file=...|jsonpath=...|jsonpath-file=...] (TYPE [NAME | -l label] | TYPE/NAME ...) [flags] [options]\n\nExamples:\n  # List all pods in ps output format.\n  oc get pods\n  \n  # List a single replication controller with specified ID in ps output format.\n  oc get rc redis\n  \n  # List all pods and show more details about them.\n  oc get -o wide pods\n  \n  # List a single pod in JSON output format.\n  oc get -o json pod redis-pod\n  \n  # Return only the status value of the specified pod.\n  oc get -o template pod redis-pod --template={{.currentState.status}}\n\nOptions:\n      --all-namespaces=false: If present, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.\n      --allow-missing-template-keys=true: If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats.\n      --experimental-use-openapi-print-columns=false: If true, use x-kubernetes-print-column metadata (if present) from openapi schema for displaying a resource.\n      --export=false: If true, use 'export' for the resources.  Exported resources are stripped of cluster-specific information.\n  -f, --filename=[]: Filename, directory, or URL to files identifying the resource to get from a server.\n      --ignore-not-found=false: Treat \"resource not found\" as a successful retrieval.\n      --include-extended-apis=true: If true, include definitions of new APIs via calls to the API server. [default true]\n  -L, --label-columns=[]: Accepts a comma separated list of labels that are going to be presented as columns. Names are case-sensitive. You can also use multiple flag options like -L label1 -L label2...\n      --no-headers=false: When using the default or custom-column output format, don't print headers (default print headers).\n  -o, --output='': Output format. One of: json|yaml|wide|name|custom-columns=...|custom-columns-file=...|go-template=...|go-template-file=...|jsonpath=...|jsonpath-file=... See custom columns [http://kubernetes.io/docs/user-guide/kubectl-overview/#custom-columns], golang template [http://golang.org/pkg/text/template/#pkg-overview] and jsonpath template [http://kubernetes.io/docs/user-guide/jsonpath].\n      --raw='': Raw URI to request from the server.  Uses the transport specified by the kubeconfig file.\n  -R, --recursive=false: Process the directory used in -f, --filename recursively. Useful when you want to manage related manifests organized within the same directory.\n      --schema-cache-dir='~/.kube/schema': If non-empty, load/store cached API schemas in this directory, default is '$HOME/.kube/schema'\n  -l, --selector='': Selector (label query) to filter on, supports '=', '==', and '!='.\n  -a, --show-all=true: When printing, show all resources (false means hide terminated pods.)\n      --show-kind=false: If present, list the resource type for the requested object(s).\n      --show-labels=false: When printing, show all labels as the last column (default hide labels column)\n      --sort-by='': If non-empty, sort list types using this field specification.  The field specification is expressed as a JSONPath expression (e.g. '{.metadata.name}'). The field in the API resource specified by this JSONPath expression must be an integer or a string.\n      --template='': Template string or path to template file to use when -o=go-template, -o=go-template-file. The template format is golang templates [http://golang.org/pkg/text/template/#pkg-overview].\n  -w, --watch=false: After listing/getting the requested object, watch for changes.\n      --watch-only=false: Watch for changes to the requested object(s), without listing/getting first.\n\nUse \"oc options\" for a list of global command-line options (applies to all commands).\n\n", "stdout": ""}}

```

##### Additional Information

- The OS is CentOS Linux release 7.4.1708 (Core)

oc status:
```
[apalhares@devops01 ~]$ oc status
In project default on server https://10.0.14.90:8443

https://docker-registry-default.router.default.svc.cluster.local (passthrough) (svc/docker-registry)
  dc/docker-registry deploys docker.io/openshift/origin-docker-registry:v3.7.0
    deployment #1 deployed 39 hours ago - 1 pod

svc/kubernetes - 172.30.0.1 ports 443->8443, 53->8053, 53->8053

https://registry-console-default.router.default.svc.cluster.local (passthrough) (svc/registry-console)
  dc/registry-console deploys docker.io/cockpit/kubernetes:latest
    deployment #1 deployed 39 hours ago - 1 pod

svc/router - 172.30.59.217 ports 80, 443, 1936
  dc/router deploys docker.io/openshift/origin-haproxy-router:v3.7.0
    deployment #10 deployed 34 minutes ago - 2 pods
    deployment #9 deployed 16 hours ago
    deployment #8 deployed 16 hours ago

View details with 'oc describe <resource>/<name>' or list everything with 'oc get all'.
[apalhares@devops01 ~]$ oc get all
NAME                            DOCKER REPO                                                 TAGS      UPDATED
imagestreams/registry-console   docker-registry.default.svc:5000/default/registry-console   latest    39 hours ago

NAME                                 REVISION   DESIRED   CURRENT   TRIGGERED BY
deploymentconfigs/docker-registry    1          1         1         config
deploymentconfigs/registry-console   1          1         1         config
deploymentconfigs/router             10         2         2         config

NAME                      HOST/PORT                                                   PATH      SERVICES           PORT      TERMINATION   WILDCARD
routes/docker-registry    docker-registry-default.router.default.svc.cluster.local              docker-registry    <all>     passthrough   None
routes/registry-console   registry-console-default.router.default.svc.cluster.local             registry-console   <all>     passthrough   None

NAME                          READY     STATUS    RESTARTS   AGE
po/docker-registry-1-2mx8w    1/1       Running   0          1d
po/registry-console-1-wbcnh   1/1       Running   0          1d
po/router-10-nht2v            1/1       Running   0          34m
po/router-10-nmv76            1/1       Running   0          34m

NAME                    DESIRED   CURRENT   READY     AGE
rc/docker-registry-1    1         1         1         1d
rc/registry-console-1   1         1         1         1d
rc/router-1             0         0         0         1d
rc/router-10            2         2         2         34m
rc/router-2             0         0         0         22h
rc/router-3             0         0         0         19h
rc/router-4             0         0         0         18h
rc/router-5             0         0         0         18h
rc/router-6             0         0         0         16h
rc/router-7             0         0         0         16h
rc/router-8             0         0         0         16h
rc/router-9             0         0         0         15h

NAME                   CLUSTER-IP       EXTERNAL-IP   PORT(S)                   AGE
svc/docker-registry    172.30.190.146   <none>        5000/TCP                  1d
svc/kubernetes         172.30.0.1       <none>        443/TCP,53/UDP,53/TCP     1d
svc/registry-console   172.30.56.146    <none>        9000/TCP                  1d
svc/router             172.30.59.217    <none>        80/TCP,443/TCP,1936/TCP   1d
```

**Comment by masterkorp** [comment]
The command being run by ansible is:
```
 /bin/oc get node --selector=region=infra --field-selector=spec.unschedulable!=true -o json -n default
```

**Comment by vrutkovs** [comment]
`oc 3.7` doesn't have ` --field-selector`. Please use `release-3.7` branch to setup 3.7 cluster, see https://github.com/openshift/openshift-docs/pull/7963

**Comment by masterkorp** [comment]
@vrutkovs the playbook `deploy_cluster`does not exist in `release-3.7` branch

**Comment by masterkorp** [comment]
@vrutkovs Another question, is that, since I am using the master branch, why is `oc 3.7` being installed?

**Comment by vrutkovs** [comment]
>the playbook deploy_clusterdoes not exist in release-3.7 branch

Right, its `playbooks/byo/config.yml`

>Another question, is that, since I am using the master branch, why is oc 3.7 being installed?

You've specified `openshift_release=3.7` in the inventory - and 3.9 is not released yet, so there's probably no repo with 3.9 code publicly available
